internal/c/libqb.cpp

```diff
@@ -37148,6 +37148,26 @@ int32 func__resize() {
     return 0;
 }
 
+void sub__windowscale(int32 amount) {
+    int32 tempx, tempy;
+        if (amount < 50) return; // don't scale the screen to less than 50% of it's original size!!
+
+        tempx = environment_2d__screen_width * amount / 100;  // basically our width of the screen
+        tempy = environment_2d__screen_height * amount / 100; // and this is the height of the screen
+        if (tempx < 320) return; // minimal window width of 320.
+        if (tempy < 240) return; // minimal window height of 240
+        
+        ScreenResize = 1;
+        ScreenResizeScale = amount;
+
+        resize_snapback = 0;
+        resize_pending = 1; // this says we're going to do a resize
+
+        glutReshapeWindow(tempx, tempy);

```

Comment:
I'm pretty sure we're not supposed to do this outside of the thread handling the GLUT logic. Will the window not get resized by the existing logic if you just wait for the next screen refresh after setting the various settings?

Reply:
Note that if you remove the GLUT logic that will also fix the compilation issues, since I assume the variables are available even in `$CONSOLE:ONLY` mode (though they wouldn't do anything). Maybe trying to use the commands should error in `$CONSOLE:ONLY` mode, but none of the other functions look like they do that so 🤷

Reply:
We're letting GLUT handle the resizing.  That's what the resize_pending = 1 is for.  At least, that's how I took it, when looking at the functions for RESIZE.  

As for the compilation issues, I wonder if there's a flag somewhere for the "Parts System" that I haven't found yet.   I'll dig into that deeper, but this should probably be a non-console command as it resizes the window and not the terminal.

Reply:
Right, GLUT does the resizing, the question is where in the code we should call `glutReshapeWindow()`. Similar to #66, it's not safe to call it from outside of the rendering logic and it may randomly have problems. It's probably not to hard to fix here, we just have to tell the rendering logic that currently does the resizing (in `GLUT_DISPLAY_REQUEST()`) what to do and have it call `glutReshapeWindow()`, rather than doing it here. There's already some `glutReshapeWindow()` calls in there that do the snapback logic and such, it would be the same but resizing the window to our scaling instead.

In regards to `ScreenResizeScale`, I ran a search over all the code but all it does is set `auto_resize` equal to it. I'm pretty sure the magic part is the `glutReshapeWindow()` call, because that triggers the `GLUT_RESHAPE_FUNC` which changes a bunch of the variables like `display_x` and friends to the new size.